src/main/java/org/apache/sysds/runtime/matrix/data/LibCommonsMath.java

```diff
@@ -295,11 +281,283 @@ private static MatrixBlock computeMatrixInverse(Array2DRowRealMatrix in) {
+		double[] eval = new double[m];
+		for(int i = 0; i < m; i++) {
+			eval[i] = in.getValue(i, i);
+		}
+
+		double[][] evec = DataConverter.convertToArray2DRowRealMatrix(Q_prod).getData();
+		return sortEVs(eval, evec);
+	}
+
+	/**
+	 * Function to compute the Householder transformation of a Matrix.
+	 *
+	 * @param in Input Matrix
+	 * @return transformed matrix
+	 */
+	private static MatrixBlock computeHouseholder(MatrixBlock in) {
+		int num_Threads = 1;
+		int m = in.rlen;
+
+		MatrixBlock A_n = new MatrixBlock(m, m, 0.0);
+		A_n.copy(in);
+
+		for(int k = 0; k < m - 2; k++) {
+			MatrixBlock ajk = A_n.slice(0, m - 1, k, k);
+			for(int i = 0; i <= k; i++) {
+				ajk.setValue(i, 0, 0.0);
+			}
+			double alpha = Math.sqrt(ajk.sumSq());
+			double ak1k = A_n.getDouble(k + 1, k);
+			if(ak1k > 0.0)
+				alpha *= -1;
+			double r = Math.sqrt(0.5 * (alpha * alpha - ak1k * alpha));
+			MatrixBlock v = new MatrixBlock(m, 1, 0.0);
+			v.copy(ajk);
+			v.setValue(k + 1, 0, ak1k - alpha);
+			RightScalarOperator op_div_scalar = new RightScalarOperator(Divide.getDivideFnObject(), 2 * r, num_Threads);
+			v = v.scalarOperations(op_div_scalar, new MatrixBlock());
+
+			MatrixBlock P = new MatrixBlock(m, m, 0.0);
+			for(int i = 0; i < m; i++) {
+				P.setValue(i, i, 1.0);
+			}
+
+			ReorgOperator op_t = new ReorgOperator(SwapIndex.getSwapIndexFnObject(), num_Threads);
+			AggregateBinaryOperator op_mul_agg = InstructionUtils.getMatMultOperator(num_Threads);
+			BinaryOperator op_add = InstructionUtils.parseExtendedBinaryOperator("+");
+			BinaryOperator op_sub = InstructionUtils.parseExtendedBinaryOperator("-");
+
+			MatrixBlock v_t = v.reorgOperations(op_t, new MatrixBlock(), 0, 0, m);
+			v_t = v_t.binaryOperations(op_add, v_t);
+			MatrixBlock v_v_t_2 = A_n.aggregateBinaryOperations(v, v_t, op_mul_agg);
+			P = P.binaryOperations(op_sub, v_v_t_2);
+			A_n = A_n.aggregateBinaryOperations(P, A_n.aggregateBinaryOperations(A_n, P, op_mul_agg), op_mul_agg);
+		}
+		return A_n;
+	}
+
+	/**
+	 * Sort the eigen values (and vectors) in increasing order (to be compatible w/ LAPACK.DSYEVR())
+	 *
+	 * @param eValues  Eigenvalues
+	 * @param eVectors Eigenvectors
+	 * @return array of matrix blocks
+	 */
+	private static MatrixBlock[] sortEVs(double[] eValues, double[][] eVectors) {
+		int n = eValues.length;
+		for(int i = 0; i < n; i++) {
+			int k = i;
+			double p = eValues[i];
+			for(int j = i + 1; j < n; j++) {
+				if(eValues[j] < p) {
+					k = j;
+					p = eValues[j];
+				}
+			}
+			if(k != i) {
+				eValues[k] = eValues[i];
+				eValues[i] = p;
+				for(int j = 0; j < n; j++) {
+					p = eVectors[j][i];
+					eVectors[j][i] = eVectors[j][k];
+					eVectors[j][k] = p;
+				}
+			}
+		}
+
+		MatrixBlock eval = DataConverter.convertToMatrixBlock(eValues, true);
+		MatrixBlock evec = DataConverter.convertToMatrixBlock(eVectors);
+		return new MatrixBlock[] {eval, evec};
+	}
+
+	private static boolean isSym(MatrixBlock in) {
```

Comment:
1. consider if you want to use it the one place it is in use i commented that you probably should not. if you decide to throw it away ignore 2.
2. The convertToArray2DRowReadlMatrix call makes this exceedingly slow, instead use the getDenseDoubleValues() to get a row major double[] that you can compare in.